### PR TITLE
fix exception

### PR DESCRIPTION
**Pull request by andersfylling**
## Checklist

- [x] Changelog entry added in `CHANGELOG` or not relevant for this change
- [x] Version updated in `pyproject.toml` or not relevant for this change

### CODE REVIEW

File: cognite/experimental/data_classes/vision.py [hunk 1]
```diff
@@ -53,8 +53,10 @@ def __init__(
         self.items = items
 
     def dump(self, camel_case: bool = False) -> Dict[str, Any]:
-        assert camel_case
-        return resource_to_camel_case(self)
+        if camel_case:
+            return resource_to_camel_case(self)
+        else:
+            return super().dump(camel_case=camel_case)
```
Comment (torbjornopheim) [inline cognite/experimental/data_classes/vision.py:59]: Shouldn't this be the other way around?

**Review by torbjornopheim — APPROVED** [review]
LGTM

File: cognite/experimental/data_classes/vision.py (hunk 1 above)
Comment (andersfylling) [reply cognite/experimental/data_classes/vision.py:59]: Yeah, it's a bit awkward here. The resource_to_camel_case is needed to deal with recursive conversion.

**Comment by torbjornopheim** [comment]
Can you point to where the string convertion of the cognite resource happens in the code? So it's clearer what needs to be patched in the base sdk? @andersfylling

**Comment by andersfylling** [comment]
@torbjornopheim I believe this is the issue: https://github.com/cognitedata/cognite-sdk-python/blob/26b24e1c34675c3e541bc7771e3bebf24b3a232d/cognite/client/data_classes/_base.py#L13

My understanding is that the first line returns either a dict or list[dict], and the json dump method does not have logic to handle dict nor list of dict which causes this to fail.

**Comment by andersfylling** [comment]
this seems to fix the main issue in the experimental sdk: https://github.com/cognitedata/cognite-sdk-python-experimental/pull/287

**Pull request closed without merging**